arbnode/batch_poster.go

```diff
@@ -982,38 +1087,87 @@ func (b *BatchPoster) encodeAddBatch(
 	} else if delayProof != nil {
 		methodName = sequencerBatchPostDelayProofMethodName
 	} else {
-		methodName = sequencerBatchPostMethodName
+		methodName = newSequencerBatchPostMethodName
 	}
 	method, ok := b.seqInboxABI.Methods[methodName]
 	if !ok {
 		return nil, nil, errors.New("failed to find add batch method")
 	}
 	var args []any
 	var kzgBlobs []kzg4844.Blob
+	var fullCalldata []byte
 	var err error
-	args = append(args, seqNum)
-	if use4844 {
-		kzgBlobs, err = blobs.EncodeBlobs(l2MessageData)
+	if methodName == newSequencerBatchPostMethodName {
+		log.Info("Coming inside the newSequencerBatchPostMethodName")
+		method, ok := b.seqInboxABI.Methods[oldSequencerBatchPostMethodName]
+		if !ok {
+			return nil, nil, errors.New("failed to find add batch method")
+		}
+		// initially constructing the calldata using the old oldSequencerBatchPostMethodName method
+		// This will allow us to get the attestation quote on the hash of the data
+		args = append(args, seqNum)
+		args = append(args, l2MessageData)
+		args = append(args, new(big.Int).SetUint64(delayedMsg))
+		args = append(args, b.config().gasRefunder)
+		args = append(args, new(big.Int).SetUint64(uint64(prevMsgNum)))
+		args = append(args, new(big.Int).SetUint64(uint64(newMsgNum)))
+
+		// Later append the delay proof if needed for getting the attestion quote.
```

Comment:
We have this comment here but are not really appending the delay proof?

Reply:
No we aren't because we are currently not using Delay Proofs. This is also the same code as the non Celestia nitro v3.5.2. Mostly I'm not sure if the delay proof should be appended to the args and sent to the function `getAttestationQuote()`

Reply:
ahh this is a good question, I think we should talk about it on zulip